### DISCUSSION TITLE
PhatomCamera3D.set_look_at_targets seems to not accept arrays containing children of Node3D

### DISCUSSION THREAD

**Discussion by AvianAnalyst** [ramokz/phantom-camera#302] [Question / Help Wanted] [answered]
![image](https://github.com/ramokz/phantom-camera/assets/45480870/44c649b7-50d2-4fcb-a17d-726071f05d90)
![image](https://github.com/ramokz/phantom-camera/assets/45480870/9eb20e1c-b2a6-4356-afe8-a6b0982f34f5)
Here's the error with debug showing the array element type, and my script. If it helps I can also show the output I get when I don't try to call set_look_at_targets that also shows the elements type. 

This seems to be saying that there's a type error when I try to set the camera look at. But this doesn't make sense to me. I'm new to godot and very open to the possibility I’m missing something obvious. I'm not new to programming but.. the error doesn't make sense to me. I would love any ideas. 

Edit to add: The error both doesn't make sense, in that it should be able to accept a derived class so I must be misreading it, and it doesn't make sense that I'd uniquely be encountering this problem, so I'm sure I'm doing something wrong.

**Comment by ramokz** [comment, marked as answer]
The error suggests that `dice` isn't of type `Array[Node3D]`. Have you tried strongly typing it?
I.e.

```gdscript
var dice: Array[Node3D] = []
```

Also, what does the two `print` statements on the second and third last line show in the output tab?

**Reply by AvianAnalyst** [reply]
![image](https://github.com/ramokz/phantom-camera/assets/45480870/7b622488-93c8-4712-947f-02c28f12f00b)
I added a type print as well. 

That said adding the type hint (and no other changes) has made it work, so thank you!

**Reply by ramokz** [reply]
No problem! Can give a quick explanation for why it worked.

Basically, `var dice = []` is the same as `var dice : Array = []`, which is **not** the same as `var dice : Array[Node3D]`.

`var dice : Array = []` can contain data beyond just `Node3D`, whereas `var dice : Array[Node3D]` can only do just that.
Because the `set_follow_targets()` function explicitly requires an array of `Node3D`, it means that an array that doesn't state it can only contain that particular data type will not pass that condition. If the requirement was “any generic array”, it would mean the function would have to sort through the array manually to only select `Node3D` entries, which it shouldn't really have to.

**Reply by AvianAnalyst** [reply]
Gotcha! Thank you!!
